docs/lua-yottadb-ydbdocs.rst

```diff
@@ -321,23 +341,66 @@ fetching the function pointer from a C closure, and using ``lua_call()``.
 
 
 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+kill (varname[, subsarray[, ...]])
```

Comment:
I realize I am reviewing documentation rather than the software, but does the latter have to use `kill()`. I realize that KILL is the name of the operation in M, but it is a bit disconcerting to those who are not insiders.

Reply:
@ksbhaskar, I'm not 100% certain of the reason behind the objection. My best guess is a concern that someone might associate this word with an action against a human.

Considerations:
- `kill` is a standard command in Unix, Windows, and M. So programmers are unlikely to mistake or associate the context with humans.
- The `kill` command does not seem to be a problem to use even with a much closer association to action on humans within, say, the entertainment industry (e.g. minecraft and worse things).
- Associated words are used commonly, even to warn of such things, without problems: e.g. "trigger warning."
- It is good to have a word that is stronger than delete because the command removes an entire tree of the database, whereas the delete action implies removal of just one node value.

Given these considerations, and its close association with the M command, I think it would be preferable to retain `kill` as at least an option for this action. Would it resolve this concern to add another alias? Considering that the action is performed on a tree, the best alternative might be `lop`.

Or have I misunderstood your concern?

Reply:
@berwynhoyt You understood my concern perfectly. OK, you articulate the case well. Let's live with kill.